src/java/org/apache/cassandra/metrics/InternodeOutboundMetrics.java

```diff
@@ -109,12 +114,41 @@ public class InternodeOutboundMetrics
     
     private final MetricNameFactory factory;
 
+    /**
+     * Factory method to create InternodeOutboundMetrics instances.
+     * If a custom provider is configured, it will be used instead of the default implementation.
+     *
+     * @param ip IP address to use for metrics label
+     * @param messagingPool the OutboundConnections instance
+     * @return an InternodeOutboundMetrics instance
+     */
+    public static InternodeOutboundMetrics create(InetAddressAndPort ip, OutboundConnections messagingPool)
+    {
+        if (CUSTOM_INTERNODE_OUTBOUND_METRICS_PROVIDER_PROPERTY.isPresent())
+        {
+            Class<InternodeOutboundMetrics> klass = FBUtilities.classForName(CUSTOM_INTERNODE_OUTBOUND_METRICS_PROVIDER_PROPERTY.getString(), "Internode Outbound Metrics Provider");
+            try
+            {
+                return klass.getDeclaredConstructor(InetAddressAndPort.class, OutboundConnections.class).newInstance(ip, messagingPool);
+            }
+            catch (NoSuchMethodException | InstantiationException | IllegalAccessException |
+                   InvocationTargetException e)
+            {
+                throw new RuntimeException(e);
```

Comment:
what happens to this ? 

do we want that consequence, or to log the error and fallback to the `InternodeOutboundMetrics` default ?

Reply:
Added tests for sonar, so now we know!